### PR TITLE
Implement draft-ietf-quic-bit-grease-02

### PR DESCRIPTION
**Pull request by Ralith**
Draft backwards-compatible anti-ossification measure.

### CODE REVIEW

File: quinn-proto/src/transport_parameters.rs [hunk 1]
```diff
@@ -381,6 +392,12 @@ impl TransportParameters {
                     }
                     params.max_datagram_frame_size = Some(r.get().unwrap());
                 }
+                0x2ab2 => {
+                    if len != 0 {
+                        return Err(Error::Malformed);
+                    }
+                    params.grease_quic_bit = true;
```
Comment (djc) [inline quinn-proto/src/transport_parameters.rs:399]: Nit: `match`?

**Review by djc — DISMISSED** [review]
Nice, LGTM.

File: quinn-proto/src/transport_parameters.rs (hunk 1 above)
Comment (Ralith) [reply quinn-proto/src/transport_parameters.rs:399]: I don't see a way to use `match` here that doesn't add more syntatic noise. Did you have something specific in mind?

Comment (djc) [reply quinn-proto/src/transport_parameters.rs:399]:
```rust
match len {
    0 => params.grease_quic_bit = true,
    _ => return Err(Error::Malformed),
}
```
?

Comment (Ralith) [reply quinn-proto/src/transport_parameters.rs]: Ah, my brain has trouble putting side-effect-oriented code in value positions like that :D

Comment (djc) [reply quinn-proto/src/transport_parameters.rs]: Statements are just expressions that yield () as far as Rust is concerned. 😎

**Review by djc — APPROVED** [review]

**Pull request merged**